### ISSUE TITLE
iocminion-pdfminer

### ISSUE THREAD

**Issue by commander1** [pun1sh3r/iocminion#1]
I have installed pdfminer and tested it by itself however Im having an issue with iocminion using it. 
pdfminer not installed see: https://github.com/euske/pdfminer 
procesing pdf...
global name 'PDFResourceManager' is not defined
Traceback (most recent call last):
  File "iocminion.py", line 373, in <module>
    main()
  File "iocminion.py", line 357, in main
    iocObj.get_hashes(data,ioc_data[args.pdf[0]]['hashes'],'\b[a-fA-F0-9]{32}\b')
  File "iocminion.py", line 123, in get_hashes
    for i in  html.findAll(text=re.compile(r'%s'% (regex),re.I)):
AttributeError: 'NoneType' object has no attribute 'findAll'

**Comment by pun1sh3r** [comment]
hey commander:) 
lets see 

open the python shell and do the following: 
shell$ python
Python 2.7.6 (default, Mar 22 2014, 22:59:56)                                                                                                                                                                                                                                  
[GCC 4.8.2] on linux2                                                                                                                                                                                                                                                          
Type "help", "copyright", "credits" or "license" for more information.                                                                                                                                                                                                         
>>> from pdfminer.pdfinterp import PDFResourceManager, process_pdf  

let me know if it imports fine. also let me know what are you trying to parse so i can have an idea for debugging

**Comment by commander1** [comment]
root@ubuntu:/home/commander/iocminion/pdfminer# python
Python 2.7.6 (default, Mar 22 2014, 22:59:38) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.


Currently I am setting up and running tests. Using the suggested PDF:
python iocminion.py  --pdf http://www.welivesecurity.com/wp-content/uploads/2015/04/mumblehard.pdf

probably user error here...I see the error handling in the script 'from pdfminer.pdfinterp import PDFResourceManager, process_pdf'

**Comment by pun1sh3r** [comment]
on the python console do you get an import error when importing:

from pdfminer.pdfinterp import PDFResourceManager, process_pdf

**Comment by commander1** [comment]
> > > from pdfminer.pdfinterp import PDFResourceManager, process_pdf
> > > Traceback (most recent call last):
> > >   File "<stdin>", line 1, in <module>
> > > ImportError: cannot import name process_pdf

**Comment by pun1sh3r** [comment]
Its possible that u might have to install process_pdf separately

**Comment by commander1** [comment]
Hmm...no python path issue?

**Comment by pun1sh3r** [comment]
Take a look at this thread
http://python.developermemo.com/11650_5742166/

**Comment by commander1** [comment]
Ok cool....good thread and on topic. That site is hosting cross-platform malware so surf safely!

**Comment by pun1sh3r** [comment]
Check the same thread on stack overflow.
http://stackoverflow.com/questions/5725278/python-help-using-pdfminer-as-a-library

**Comment by pun1sh3r** [comment]
were u able to solve the problem?

**Comment by commander1** [comment]
Sorry no been traveling and my python is more searching cutting and pasting and moving things around until they work.

**Comment by commander1** [comment]
Awesome....works great

**Comment by pun1sh3r** [comment]
Awesome man. Let me know if there is any features you would like to see ..

**Comment by commander1** [comment]
hmm, so I was working with the gmail function. I had to change to this for it to actually see unread and parse the email. 
 87       try:
 88       emailData = defaultdict(set)
 89       g = gmail.login(username,passwd)
 90       if(g.logged_in):
 91         print "sucessful login..."
 92         today = datetime.date.today().strftime("%Y-%m-%d")
 93         emails = g.inbox().mail(unread=True)

**Comment by commander1** [comment]
need it to find hxxp://................
which I will add to the def **init**(self):
or would the regex for -->   self.domainRegex find it? Seems it isn't parsing domains correctly
but your were using TLD's which is better
Also trying to think of a way to add context to IOC's parsed

**Comment by pun1sh3r** [comment]
i didnt quite understood what you meant on your previous email.  can you
please elaborate :)

**Comment by commander1** [comment]
Apologies....So I have it checking into email and parsing out the IOC's. It is just being picky. Not sure why they do not get parsed. Which I get....needs to be tested, tweaked and fined tuned for the results I'm looking for. Many of the email iocs looks like this:
- 33.155.4.90:12388 - GET /TACH3/
- 44.20.133.163 port 80
  --> hxxp://conueniveritiswhitelist.divadevaspa[.]com/coauthored/viewforu
  Payload URLs:
  api.ge[.]rt/1/files/4gypyaK2/0/blob

Your def **init**(self):
    self.md5Regex = "\b[a-fA-F0-9]{32}\b"
    self.sha1Regex = "\b[a-fA-F0-9]{40}\b"
    self.etc
    self.etc 
 should hit on all of the above correct?

**Comment by pun1sh3r** [comment]
yeah it should parse correctly.   very odd..  because i use the email
function almost every day without issues

**Comment by pun1sh3r** [comment]
were you able to solve the issue with the email stuffz?
